src/components/inputs/Select/index.js

```diff
@@ -1,123 +1,207 @@
     errors,
     onChange,
     options,
+    touched,
     onBlur,
-    onFocus,
-    withMarginTop,
     multi,
     disabled,
     clearable,
-    isFocused,
-    searchable,
     required,
-    classes,
-    // noResultsText,
+    noOptionsText,
+    getOptionLabel,
+    getOptionSelected,
+    loading,
+    renderOption,
 }) => {
-    const [selectInputValue, setSelectInputValue] = useState('');
-    const hasErrors = errors.length > 0;
-    const classNames = hasErrors
-        ? [classes.select, classes.selectError]
-        : [classes.select];
+    const [selectedValue, setSelectedValue] = useState(null);
+    const [multiSelectedValue, setMultiSelectedValue] = useState([]);
     const intl = useSafeIntl();
-    const [focus, setFocus] = useState(isFocused);
+    const classes = useStyles();
+
+    const getOption = optionValue =>
+        options.find(o => `${o.value}` === `${optionValue}`);
+
+    useEffect(() => {
+        if (value) {
+            if (multi) {
+                const valuesList = Array.isArray(value)
+                    ? value
+                    : value.split(',');
+                const newSelectedValue = valuesList
+                    .map(v => getOption(v))
+                    .filter(o => o);
+                setMultiSelectedValue(newSelectedValue);
+            } else {
+                const newSelectedValue = getOption(value);
+                if (newSelectedValue) setSelectedValue(newSelectedValue);
+            }
+        } else {
+            multi ? setMultiSelectedValue([]) : setSelectedValue(null);
+        }
+    }, [value, options, multi]);
+
+    const handleChange = useCallback(
+        (e, newValue) => {
+            if ((!multi && !newValue) || (multi && newValue.length === 0)) {
+                return onChange(null);
+            }
+            if (multi) {
+                return onChange(newValue.map(v => v && v.value).join(','));
+            }
+            return onChange(newValue.value);
+        },
+        [multi, onChange],
+    );
+    const extraProps = {
+        getOptionLabel: getOptionLabel || (option => option && option.label),
+        getOptionSelected:
+            getOptionSelected ||
+            ((option, val) => val && option.value === val.value),
+    };
+    if (renderOption) {
+        extraProps.renderOption = renderOption;
+    }
+
+    const renderInput = params => {
+        const paramsCopy = {
+            ...params,
+        };
+        let inputExtraProps = {};
+        if (extraProps.renderOption && params.inputProps.value) {
+            inputExtraProps = {
+                startAdornment: (
+                    <div className={classes.startAdornment}>
+                        {extraProps.renderOption({
+                            label: params.inputProps.value,
+                        })}
+                    </div>
+                ),
+                style: { color: 'transparent' },
+            };
+            inputExtraProps.startAdornment = (
```

Comment:
probably the wrong key name, as it the same as same as the one just above?

Reply:
the second one is useless